hw/src/lib.rs

```diff
@@ -260,9 +259,11 @@ impl HardwareWalletManager {
 impl Drop for HardwareWalletManager {
 	fn drop(&mut self) {
 		self.exiting.store(true, atomic::Ordering::Release);
-		if let Some(thread) = self.update_thread.take() {
-			thread.thread().unpark();
-			thread.join().ok();
+		for thread in self.active_threads.iter_mut() {
```

Comment:
better to kill the source of a potential hang for now. what are the repercussions if the process exits _without_ joining the threads?

Reply:
omg. yes now I get your point the threads will never terminate and consequently this will hang. 

If the ```main thread``` just terminates the two other threads will run and listen for new events and try to connect/disconnect to inserted trezors and ledgers. But I think that is ok since these threads as suppose to live as long as parity does!

Otherwise I think a message passing channel would be appropriate here and signal the threads because rust doesn't provide ```thread cancel``` as pthread does.

Thoughts?

Reply:
I think signalling that Parity has gone down with an atomic bool or similar is fine. Just that I don't think blocking while waiting for them to join is necessarily a good idea.